### ISSUE TITLE
Add support for decrypting the traffic

### ISSUE THREAD

**Issue by tsg** [elastic/beats#161]
We could have partial support for decrypting traffic by having the user provide the certificate / private keys.

This would be especially useful for PostgreSQL where encryption is enabled by default. We should also support HTTPS decryption under the same conditions.

This won't work in the outbound case (when calling a 3rd party API) but it can still be useful in many cases.

**Comment by owulff** [comment]
+1

**Comment by navenel** [comment]
+1

**Comment by rswart** [comment]
I am skeptic about this. I have clients that are capturing HTTPS using other solutions but they are running issues now that Diffie-Hellman based key exchange algorithms are starting to dominate (TLS 1.2). They all run into the issue that it is 'impossible' to decrypt sessions that use DH during key exchange .

An alternative is terminating the TLS session on a load balancer or proxy and (optionally) setting up a new HTTPS session to the backend.

**Comment by w-vi** [comment]
I'd state in docs that decryption works only on rsa based key exchange an leave it up to the user what to do with some ptions howto set it up. From my point of view it still makes sense to suport it.

**Comment by agirbal** [comment]
This would be extremely useful. Most of our packets use TLS over TCP and thus makes everything opaque besides the handshake.
Many softwares allow you to specify either the cert/key (which is pretty much useless today with DHE) or a master key file (in NSS format).
Wireshark / tshark allow you to do this for example. Unfortunately they don't allow you to save back the packets as a pcap-like file, that then could be ingested with packetbeat.
Bottom line: it'd be really awesome if packetbeat could just take the nss file and attempt TLS decryption!
https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS/Key_Log_Format
